application/clicommands/DownloadCommand.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Icinga\Module\Reporting\Clicommands;
+
+use Exception;
+use Icinga\Module\Pdfexport\ProvidedHook\Pdfexport;
+use Icinga\Module\Reporting\Cli\Command;
+use Icinga\Module\Reporting\Report;
+
+class DownloadCommand extends Command
+{
+    /**
+     * Download report using format types pdf|csv|json (default pdf)
+     *
+     * USAGE:
+     *
+     *   icingacli download <id> --format
+     *
+     */
+    public function indexAction()
+    {
+        $id = $this->params->shiftRequired('id');
+
+        $report = Report::fromDb($id);
+        $format = $this->params->get('format', 'pdf');
+        $name = sprintf(
+            '%s (%s) %s',
+            $report->getName(),
+            $report->getTimeframe()->getName(),
+            date('Y-m-d H:i')
+        );
+
+        switch ($format) {
+            case 'pdf':
+                $content = Pdfexport::first()->htmlToPdf($report->toPdf());
+                break;
+            case 'csv':
+                $content = $report->toCsv();
+                break;
+            case 'json':
+                $content = $report->toJson();
+                break;
+            default:
+                throw new Exception('Format is invalid');
```

Comment:
`throw new InvalidArgumentException(sprintf('Format %s is not supported', $format));`

Unsupported is friendlier than invalid, and it shows exactly what is wrong.